meson.build

```diff
@@ -595,21 +594,9 @@ foreach decl : ['char16_t',
                 endif
         endif
 
-        if decl == 'struct statx'
-                if have
-                        want_linux_stat_h = false
-                else
-                        have = cc.sizeof(decl,
-                                         prefix : decl_headers + '#include <linux/stat.h>',
-                                         args : '-D_GNU_SOURCE') > 0
-                        want_linux_stat_h = have
-                endif
-        endif
-
```

Comment:
can we really drop all statx compat glue? i mean, there are multiple versions of statx around, and we need the one with mnt id subvol id. is our glibc+kernel baseline guaranteed to carry that version at the very least?

Reply:
sys/stat.h includes linux/stat.h. Hence, we can safely drop all.

Reply:
Added a test case about that.